src/include/txn_inline.h

```diff
@@ -705,30 +705,65 @@ __txn_visible_id(WT_SESSION_IMPL *session, uint64_t id)
 }
 
 /*
- * __wt_txn_visible --
- *     Can the current transaction see the given ID / timestamp?
+ * __wt_txn_timestamp_visible --
+ *     Can the current transaction see the given timestamp?
  */
 static inline bool
-__wt_txn_visible(WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t timestamp)
+__wt_txn_timestamp_visible(WT_SESSION_IMPL *session, wt_timestamp_t timestamp)
 {
     WT_TXN *txn;
     WT_TXN_SHARED *txn_shared;
 
     txn = session->txn;
     txn_shared = WT_SESSION_TXN_SHARED(session);
 
-    if (!__txn_visible_id(session, id))
+    /* Timestamp check. */
+    if (!F_ISSET(txn, WT_TXN_SHARED_TS_READ) || timestamp == WT_TS_NONE)
+        return (true);
+
+    return (timestamp <= txn_shared->read_timestamp);
+}
+
+/*
+ * __wt_txn_snap_min_visible --
+ *     Can the current transaction snapshot minimum/read timestamp see the given ID/timestamp? This
+ *     visibility check should only be used when assessing broader visibility based on aggregated
+ *     time window. It does not reflect whether a specific update is visible to a transaction.
+ */
+static inline bool
+__wt_txn_snap_min_visible(
+  WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t timestamp, wt_timestamp_t durable_timestamp)
+{
+    WT_UNUSED(durable_timestamp); /* Not needed since 6.0 doesn't support checkpoint cursors. */
```

Comment:
```suggestion
    /* Not needed since 6.0 doesn't support checkpoint cursors. */
    WT_UNUSED(durable_timestamp);
```

Reply:
I prefer to remove the variable `durable_timestamp` completely.

Reply:
I left it in so that future backports using this function don't have to have the argument removed. I can remove it if you don't think this is worth worrying about.

Reply:
Okay, got it. I am fine with leaving it.

Reply:
@etienneptl  yep - I tried both `test_checkpoint30` and `test_bug032`, both failed on 6.0 and passed on this branch. Didn't try the test/format configuration since the Python tests looked like more reliable repros.